sympy/vector/lamecoeff.py

```diff
@@ -0,0 +1,58 @@
+from sympy.core.basic import Basic
+from sympy import sqrt, sin, cos
+from sympy.core.function import Derivative
+
+
+class CoeffProvider(Basic):
+    """
+    It provides specific information of curvilinear coordinate
+    system to CoordSysCartesian.
+
+    Ideally, users should not instantiate this class.
+
+    """
+
+    def __new__(cls, system, curv_coord_name):
```

Comment:
I would not pass the `system` variable, if this class has to be associated to a single coordinate system, then why not merge the code in `CoordSysCartesian`? This class only makes sense as a separate class if you want to shared the single instance between multiple coordinate systems.